SmokeTests/spm/Makefile

```diff
@@ -0,0 +1,45 @@
+.PHONY: clean install test create-src-from-xcodeproj
+
+REPO_ROOT := ../../
+include ../../tools/utils/common.mk
+
+ifeq ($(PLATFORM), iOS Simulator)
```

Comment:
/comment

IIRC, we updated SPM tests with macOS support, it not then we should include  macOS and visionOS.

If you can do along with this PR :great else we can do as separate PT too.

Reply:
It is already there, in this PR 👍. We distinguish:
- **smoke tests** - integrate SDK and run basic UI test (launch the app → check no crash)
- **spm build tests** - build SDK for all supported platforms

The `SmokeTest/spm/Makefile` in question performs smoke testing. It integrates SDK into `SPMProject` via Xcode's SPM (using current branch), then launches it to assert there is no runtime crash. We do it for iOS and tvOS.

The `spm.sh` added earlier is renamed to `spm-build.sh` and ran from `.gitlab-ci.yml`:
```
make spm-build-ios
make spm-build-tvos
make spm-build-visionos
make spm-build-macos
```
It validates that `Package.swift` builds for all these platforms. Both iOS and tvOS builds might be considered redundant considering `SmokeTest/spm/Makefile`, but I wanted to avoid changing existing concepts.